src/Processors/Formats/Impl/AvroRowInputFormat.cpp

```diff
@@ -644,11 +644,21 @@ class AvroConfluentRowInputFormat::SchemaRegistry
                 request.setHost(url.getHost());
 
                 auto session = makePooledHTTPSession(url, timeouts, 1);
-                session->sendRequest(request);
-
-                Poco::Net::HTTPResponse response;
-                auto * response_body = receiveResponse(*session, request, response, false);
+                std::istream * response_body{};
+                try
+                {
+                    session->sendRequest(request);
 
+                    Poco::Net::HTTPResponse response;
+                    response_body = receiveResponse(*session, request, response, false);
+                }
+                catch (const Poco::Exception & e)
+                {
+                    /// We use session data storage as storage for exception text
+                    /// Depend on it we can deduce to reconnect session or reresolve session host
```

Comment:
How it will be re-resolved?

Reply:
See #5534

https://github.com/ClickHouse/ClickHouse/blob/93d21e0764f8a4961bd94c944fdd5f2f0bda07e7/src/IO/HTTPCommon.cpp#L172-L182

So next getSession call will see that exception happened and will reresolve. 

Hacky, but it already used in other place (so we can assume it works) and inventing a cleaner solution is not simple and that would require extra testing and more intrusive fix (maybe even patching Poco).